engine/src/main/java/org/terasology/network/internal/NetworkSystemImpl.java

```diff
@@ -868,6 +900,18 @@ private void connectClient(Client client) {
         for (BlockFamily registeredBlockFamily : blockManager.listRegisteredBlockFamilies()) {
             serverInfoMessageBuilder.addRegisterBlockFamily(registeredBlockFamily.getURI().toString());
         }
+        Path homePath = PathManager.getInstance().getHomePath();
```

Comment:
Note that this homePath is probably going to be different from the homePath cached in ServerConnectListManager on class load.   I'm surprised you haven't seen problems due to this.